domain/application/state/application_endpoint.go

```diff
@@ -766,3 +797,146 @@ WHERE  uuid = $applicationID.uuid
 
 	return endpoints, nil
 }
+
+type BindingType string
+
+const (
+	BindingRelation BindingType = "relation"
+	BindingExtra    BindingType = "extra"
+)
```

Comment:
Since this is a db type, I don't think we should use a string alias. But since this doesn't leave the state layer, I'm not sure

What do you think @manadart ?